### PR TITLE
DM-20994: Write how to make gen3 obs package guide

### PR DESCRIPTION
**Pull request by parejkoj**

### CODE REVIEW

File: doc/lsst.obs.base/creating-an-obs-package.rst [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+
+Run this test via
+
+.. code-block:: bash
+
+    pytest -sv tests/test_instrument.py
+
+the tests should fail, as there is no Example `Instrument` yet.
+
+Next, add a file in ``python/lsst/obs/example/instrument.py`` containing a subclass of `Instrument`, `ExampleCam`, which at minimum overrides these abstract methods: `Instrument.getName`, `Instrument.getCamera`, `Instrument.register`, `Instrument.filterDefinitions`, and define ``self.configPaths`` in ``__init__``.
```
Comment (timj) [inline doc/lsst.obs.base/creating-an-obs-package.rst]: Is the `ExampleCam` here the name of the `Instrument` subclass? I'd expect that to be `ExampleCamInstrument` (and double backticks).

Comment (parejkoj) [reply doc/lsst.obs.base/creating-an-obs-package.rst]: For HSC and DECam, we named them "HyperSuprimeCam" and "DarkEnergyCamera". I see you did `LsstCamInstrument` for LSST. Should we change the former two?

Comment (timj) [reply doc/lsst.obs.base/creating-an-obs-package.rst]: I think I would understand it better if the parenthetical said ", named ``ExampleCam``,"

Comment (parejkoj) [reply doc/lsst.obs.base/creating-an-obs-package.rst]: Do we want to go with `ExampleCamInstrument` here and update HSC/DECam? Now's our chance to enshrine the "correct" way for the future. I don't have a strong opinion one way or the other.

File: doc/lsst.obs.base/creating-an-obs-package.rst [hunk 2]
```diff
@@ -0,0 +1,188 @@
+.. py:currentmodule:: lsst.obs.base
+
+.. creating-an-obs-package:
+
+####################################
+How to create a new LSST obs package
+####################################
+
+These instructions describe how to create a package that allows the LSST Science Pipelines to read raw data from a telescope and instrument using the gen3 middleware and ingest it into an LSST repository.
+"Ingestion" is the process of reading raw files, interpreting their metadata, and formatting the data in a way the :py:mod:`Data Butler <lsst.daf.butler>` can read.
```
[the remark this answers is not in the document]
Comment (parejkoj) [reply doc/lsst.obs.base/creating-an-obs-package.rst]: Good point: I was never quite clear on what exactly happens with the Formatter during Ingest. Could you please offer an alternate wording?

**Comment by parejkoj** [comment]
> Is the intent to also include examples of how to write the instrument and rawformatter classes themselves?

I do have the `ExampleFormatter` class in the Formatter section: it is very short, as it doesn't need to have much in it. I did not yet put in the `ExampleInstrument` class because that one is still in flux as I finish up DECam.

File: doc/lsst.obs.base/creating-an-obs-package.rst (hunk 2 above)
[the remark this answers is not in the document]
Comment (timj) [reply doc/lsst.obs.base/creating-an-obs-package.rst:10]: "...interpreting their metadata, recording how to read the data, and copying it into the butler datastore" ?

**Pull request merged**